src/kernels/common/commonFinder.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { Memento } from 'vscode';
+import { noop } from '../../platform/common/utils/misc';
+
+// Two cache keys so we can get local and remote separately
+export const LocalKernelSpecsCacheKey = 'JUPYTER_LOCAL_KERNELSPECS_V4';
+export const RemoteKernelSpecsCacheKey = 'JUPYTER_REMOTE_KERNELSPECS_V4';
+
+export async function removeOldCachedItems(globalState: Memento): Promise<void> {
+    await Promise.all(
+        [
+            'JUPYTER_LOCAL_KERNELSPECS',
+            'JUPYTER_LOCAL_KERNELSPECS_V1',
+            'JUPYTER_LOCAL_KERNELSPECS_V2',
+            'JUPYTER_LOCAL_KERNELSPECS_V3',
+            'JUPYTER_REMOTE_KERNELSPECS',
+            'JUPYTER_REMOTE_KERNELSPECS_V1',
+            'JUPYTER_REMOTE_KERNELSPECS_V2',
+            'JUPYTER_REMOTE_KERNELSPECS_V3'
+        ]
+            .filter((key) => LocalKernelSpecsCacheKey !== key && RemoteKernelSpecsCacheKey !== key) // Exclude latest cache key

```

Comment:
Do you need this? The most recent V4 version strings are not in the list currently, so I don't think this filter is doing anything unless it's for future safety.

Reply:
We used to have this so I kept it still, not sure if we can remove it now.

Reply:
Yeah, that's the tricky bit with refactors, hard to see in the PR what is new versus a move. I doesn't seem harmful so I guess remove / keep at your discretion.